specutils/io/default_loaders/tabular_fits.py

```diff
@@ -69,6 +68,10 @@ def tabular_fits_loader(file_name, column_mapping=None, hdu=1, **kwargs):
 
     tab = Table.read(file_name, format='fits', hdu=hdu)
 
+    # Minimal consistency check for wcs consistency with table data
+    if wcs.naxis != 1:
```

Comment:
It's not always guaranteed that the number of world coordinate axes in the fits file will be one (e.g. a data cube, or data with a time dimension). Perhaps it'd be better to check explicitly for a spectral axis?

Reply:
I was not sure to what extent `tabular-fits` does, or is supposed to, support >1D spectra (thought that `generic_cube` was for that case, but again I don't have any sample files to test). But if any,  I would rather expect it to support the simplest case of a spectral cube or time series with a common (single 1D) spectral axis. Other cases, where each flux column could have its own zero point or completely individual dispersion scale, or possible intermediate cases where some subset of the spectra shares a spectra axis, are IMO really better left to `generic_cube`; at least I have no idea how to correctly identify such cases where

1. the flux data have >1 dimension

2. the spectra in the flux cube do not have identical spectral axes and

3. these wavelength/frequency scales are not included in the `BINTABLE`, but must be constructed from a WCS defined in the header.

Perhaps @teuben who wrote the initial version of `generic_cube` has a better idea about what possible data formats could be encountered.

So I would tend to just check for
```
    if not (wcs.naxis == 1 and wcs.array_shape[0] == len(tab):
```
which could be extended to
```
    if not (wcs.naxis == 1 and wcs.array_shape[0] == len(tab) or
            wcs.array_shape == tab[fluxname].shape):
```
just to allow for the case with all individual spectral axes, but note that then the correct column for the flux would have to be identified from either `column_mapping` or analogously to `_find_spectral_column`.

Reply:
Hmm. This is tricky. Ideally, the wcs should _tell_ us the column axis type (e.g. `wcs.world_axis_names`, `wcs.world_axis_types`, etc). But obviously this requires that the fits standard be well followed in all data files (😆). Maybe @eteq has some insight.

I'm tempted to just say "yeah, you're right -- people should just using `generic_cube`), but if that's the case, we should raise an appropriate error when `wcs.naxis != ` to point users to `generic_cube` for those cases.

Reply:
Problem is also I cannot even tell if `generic_cube` works, as I don't really know how a valid file should look like.